packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -328,6 +328,51 @@ describe('KeyringController', () => {
         });
       });
     });
+
+    // Testing fix for bug #4157 {@link https://github.com/MetaMask/core/issues/4157}
+    it('should return an existing HD account if the accountCount is lower than oldAccounts', async () => {
```

Comment:
Now that #4182 has been merged, I'm not sure this test case will work:

`KeyringController.getAccounts` has been refactored to return what's in the controller state, instead of keyring accounts. This also means that if accounts are added directly on a keyring without calling `persistAllKeyrings`, you won't be able to find them in the `keyringController.getAccounts()` response.

Reply:
Hmm looking at this better, it will look like `await mockKeyring.addAccounts(2);` actually added two accounts, but that's just because we previously stubbed the class with them, so the call is probably superfluous